json/json-core/src/main/scala/io/sphere/json/FromJSON.scala

```diff
@@ -391,8 +391,10 @@ object FromJSON extends FromJSONInstances {
       .appendPattern("'T'[HH[:mm[:ss]]]")
       .appendFraction(time.temporal.ChronoField.NANO_OF_SECOND, 0, 9, true)
       .optionalStart()
-      .appendOffset("+HHmm", "Z")
+      .appendOffset("+HH:MM","Z")
```

Comment:
The [javadoc](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatterBuilder.html#appendOffset-java.lang.String-java.lang.String-) for this `appendOffset` method is a bit confusing. First off, a valid zone offset needs to be one of the following (the upper case M has nothing to do with month-of-year, offset formats seemingly have their own namespace):

```
+HH - hour only, ignoring minute and second
+HHmm - hour, with minute if non-zero, ignoring second, no colon
+HH:mm - hour, with minute if non-zero, ignoring second, with colon
+HHMM - hour and minute, ignoring second, no colon
+HH:MM - hour and minute, ignoring second, with colon
+HHMMss - hour and minute, with second if non-zero, no colon
+HH:MM:ss - hour and minute, with second if non-zero, with colon
+HHMMSS - hour, minute and second, no colon
+HH:MM:SS - hour, minute and second, with colon
```

I think "with minute if non-zero" part has to be read as "with optional minutes". The following String "2024-10-14T07:00:45+00" (only hours provided as offset) can be parse using this offset format `+HHmm` , but it will fail when using `+HHMM`. The same String could also be parsed using `+HH:mm`.

Another way of writing the two optional offset block at once would have been `.appendPattern("[XXX][X]")`, but the way used here seemed a bit more descriptive to me.